src/orders/src/main/java/mushop/orders/services/OrdersService.java

```diff
@@ -0,0 +1,123 @@
+package mushop.orders.services;
+
+import mushop.orders.config.OrdersConfigurationProperties;
+import mushop.orders.entities.*;
+import mushop.orders.repositories.CustomerOrderRepository;
+import mushop.orders.resources.NewOrderResource;
+import mushop.orders.values.OrderUpdate;
+import mushop.orders.values.PaymentRequest;
+import mushop.orders.values.PaymentResponse;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.core.ParameterizedTypeReference;
+import org.springframework.stereotype.Service;
+
+import java.io.IOException;
+import java.util.Calendar;
+import java.util.List;
+import java.util.concurrent.*;
+
+import static mushop.orders.controllers.OrdersController.OrderFailedException;
+import static mushop.orders.controllers.OrdersController.PaymentDeclinedException;
+
+@Service
+public class OrdersService {
+
+    private final Logger LOG = LoggerFactory.getLogger(getClass());
+    @Autowired
+    private OrdersConfigurationProperties config;
+
+    @Autowired
+    private AsyncGetService asyncGetService;
+
+    @Autowired
+    private MessagingService messagingService;
+
+    @Autowired
+    private CustomerOrderRepository customerOrderRepository;
+
+    @Value(value = "${http.timeout:5}")
+    private long timeout;
+
+    private ScheduledExecutorService cartDeleteExecutor = Executors.newScheduledThreadPool(5);
+
+
+    public CustomerOrder createNewOrder(NewOrderResource orderPayload) {
+        LOG.info("Creating order {}", orderPayload);
+        LOG.debug("Starting calls");
+        try {
+            Future<Address> addressFuture = asyncGetService.getObject(orderPayload.address,
+                    new ParameterizedTypeReference<Address>() {
+                    });
+            Future<Customer> customerFuture = asyncGetService.getObject(orderPayload.customer,
+                    new ParameterizedTypeReference<Customer>() {
+                    });
+            Future<Card> cardFuture = asyncGetService.getObject(orderPayload.card,
+                    new ParameterizedTypeReference<Card>() {
+                    });
+            Future<List<Item>> itemsFuture = asyncGetService.getDataList(orderPayload.items,
+                    new ParameterizedTypeReference<List<Item>>() {
+                    });
+            LOG.debug("End of calls.");
+
+            //Calculate total
+            float amount = calculateTotal(itemsFuture.get(timeout, TimeUnit.SECONDS));
+
+            // Call payment service to make sure they've paid
+            PaymentRequest paymentRequest = new PaymentRequest(
```

Comment:
This payment request was tested against the payment service? The request JSON have more than expected fields. Please document for future use on the payments service.

Reply:
The payload has not changed - I just refactored the code make the call from the controller to making the call from a service. Its been tested, and works fine.